pkg/apiserver/registry/datapackaging/package_crd_rest.go

```diff
@@ -385,9 +380,19 @@ func (r *PackageCRDREST) deleteGlobalPackagesFromNS(ctx context.Context, ns stri
 	return err
 }
 
-func (r *PackageCRDREST) format(in string) string {
-	if len(in) > 50 {
-		return in[:47] + "..."
+func (r *PackageCRDREST) applySelector(list *datapackaging.PackageList, selector fields.Selector) *datapackaging.PackageList {
+	if selector == nil || selector.Empty() {
+		return list
 	}
-	return in
+
+	filteredPVs := []datapackaging.Package{}
+	for _, pv := range list.Items {
+		fieldSet := fields.Set{"spec.packageMetadataName": pv.Spec.PackageMetadataName, "metadata.name": pv.Name, "metadata.namespace": pv.Namespace}
+		if selector.Matches(fieldSet) {
```

Comment:
spec.packageMetadataName -> spec.refName

Reply:
This will need to be updated in the CRD definitions as well. Not sure if thats done already

Reply:
Currently we aren't generating anything for the associated crds, but I am sure they can be.